torchgen/gen_executorch.py

```diff
@@ -564,9 +580,31 @@ def translate_native_yaml(
                     opname = "aten::" + opname
                 assert opname in schema_dict
                 e["func"] = schema_dict.get(opname)
+
+                # Write out kernel information
+                if opname in kernel_dict:
+                    e["kernels"] = kernel_dict.get(opname)
+
```

Comment:
My understanding for why you need to comment out the assertion is that you are writing these fields into the yaml file and torchgen doesn't recognize it. To avoid this issue you don't have to write it into the file, instead you can just return it. Or move the parsing logic out of this function.

Reply:
Not entirely, while writing out would results in the assertion failing in parse_yaml_files (caller of translate_native_yaml), the assertion would still fail on the initial parse of "native_functions.yaml"

Not writing it to the out file (translated_yaml) doesn't resolve the initial parse

Reply:
So you need to pop it out in translate()